### ISSUE TITLE
Preview 13 failing to launch at random

### ISSUE THREAD

**Issue by neilunderwood** [digimezzo/dopamine-windows#1029]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ x ] Bug report  
[ ] Feature request
</code></pre>

## I'm using these versions

* Windows version: 18362.295
* Dopamine version: Windows reports 2.0.0.106, but Dopamine log states 2.0.0.1013

## Reproduction steps

* Step 1- attempt to launch Dopamine
* Step 2 - nothing happens
* Step 3 - try it again -still nothing

## Observed behavior

Sometimes it launches; most of the time it does not.  If I delete the %appdata%\Dopamine folder, the next time I launch Dopamine it launches and goes through initial setup just fine.  Next time I launch it (usually after a reboot) the application fails to launch.

## Desired behavior

It should open and retain my settings.

## Log file

##Looking at this it appears there is an issue with the location of the "Dopamine\Playlists" folder
##This is a clean log file.  I deleted it then attempted one time to launch Dopamine

2019-09-10 10:15:34.703|Error|PlaylistService..ctor|54|Could not create Playlists folder. Exception: Access to the path 'Dopamine\Playlists' is denied.
2019-09-10 10:15:34.725|Error|App.xaml.ExecuteEmergencyStop|541|Unhandled Exception. Exception:
System.Windows.Markup.XamlParseException: Set property 'Prism.Mvvm.ViewModelLocator.AutoWireViewModel' threw an exception. ---> System.ArgumentException: The directory name Dopamine\Playlists is invalid.
   at System.IO.FileSystemWatcher.set_Path(String value)
   at Dopamine.Core.Helpers.GentleFolderWatcher..ctor(String folderPath, Boolean includeSubdirectories, Int32 intervalMilliSeconds)
   at Dopamine.Services.Playlist.PlaylistService..ctor(ITrackRepository trackRepository, IFileService fileService, IContainerProvider container)
   at (Closure\`2 , Object[] , IResolverContext , IScope )
   at DryIoc.SingletonScope.GetOrAddItem(Object[] bucket, Int32 index, CreateScopedValue createValue)
   at DryIoc.Factory.ApplyReuse(Expression serviceExpr, IReuse reuse, Boolean tracksTransientDisposable, Request request)
   at DryIoc.Factory.GetExpressionOrDefault(Request request)
   at DryIoc.Factory.GetDelegateOrDefault(Request request)
   at DryIoc.Container.ResolveAndCacheDefaultDelegate(Type serviceType, Boolean ifUnresolvedReturnDefault)
   at Prism.DryIoc.Ioc.DryIocContainerExtension.Resolve(Type type)
   at Prism.Ioc.IContainerProviderExtensions.Resolve[T](IContainerProvider provider)
   at Dopamine.ViewModels.Common.Base.ContextMenuViewModelBase..ctor(IContainerProvider container)
   at Dopamine.ViewModels.Common.Base.CommonViewModelBase..ctor(IContainerProvider container)
   at Dopamine.ViewModels.Common.Base.TracksViewModelBase..ctor(IContainerProvider container)
   at Dopamine.ViewModels.Common.NowPlayingControlViewModel..ctor(IContainerProvider container)
   at (Object[] , IResolverContext , IScope )
   at DryIoc.Container.ResolveAndCacheDefaultDelegate(Type serviceType, Boolean ifUnresolvedReturnDefault)
   at Prism.DryIoc.Ioc.DryIocContainerExtension.ResolveViewModelForView(Object view, Type viewModelType)
   at Prism.Mvvm.ViewModelLocationProvider.AutoWireViewModelChanged(Object view, Action\`2 setDataContextCallback)
   at Prism.Mvvm.ViewModelLocator.AutoWireViewModelChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
   at System.Windows.DependencyObject.OnPropertyChanged(DependencyPropertyChangedEventArgs e)
   at System.Windows.FrameworkElement.OnPropertyChanged(DependencyPropertyChangedEventArgs e)
   at System.Windows.DependencyObject.NotifyPropertyChange(DependencyPropertyChangedEventArgs args)
   at System.Windows.DependencyObject.UpdateEffectiveValue(EntryIndex entryIndex, DependencyProperty dp, PropertyMetadata metadata, EffectiveValueEntry oldEntry, EffectiveValueEntry& newEntry, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType)
   at System.Windows.DependencyObject.SetValueCommon(DependencyProperty dp, Object value, PropertyMetadata metadata, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType, Boolean isInternal)
   at System.Windows.Baml2006.WpfMemberInvoker.SetValue(Object instance, Object value)
   at MS.Internal.Xaml.Runtime.ClrObjectRuntime.SetValue(Object inst, XamlMember property, Object value)
   --- End of inner exception stack trace ---
   at System.Windows.Markup.XamlReader.RewrapException(Exception e, IXamlLineInfo lineInfo, Uri baseUri)
   at System.Windows.Markup.WpfXamlLoader.Load(XamlReader xamlReader, IXamlObjectWriterFactory writerFactory, Boolean skipJournaledProperties, Object rootObject, XamlObjectWriterSettings settings, Uri baseUri)
   at System.Windows.Markup.WpfXamlLoader.LoadBaml(XamlReader xamlReader, Boolean skipJournaledProperties, Object rootObject, XamlAccessLevel accessLevel, Uri baseUri)
   at System.Windows.Markup.XamlReader.LoadBaml(Stream stream, ParserContext parserContext, Object parent, Boolean closeStream)
   at Dopamine.Views.MiniPlayer.MiniPlayerPlaylist..ctor(Windows10BorderlessWindow parent, IPlaybackService playbackService, IRegionManager regionManager, IEventAggregator eventAggregator)
   at (Closure\`3 , Windows10BorderlessWindow )
   at Dopamine.Views.Shell.InitializeWindows()
   at Dopamine.Views.Shell..ctor(IContainerProvider container, IWindowsIntegrationService windowsIntegrationService, II18nService i18nService, INotificationService notificationService, IWin32InputService win32InputService, IAppearanceService appearanceService, IPlaybackService playbackService, IMetadataService metadataService, ILifetimeService lifetimeService, IEventAggregator eventAggregator)
   at (Closure\`8 , Object[] , IResolverContext , IScope )
   at DryIoc.Container.ResolveAndCacheDefaultDelegate(Type serviceType, Boolean ifUnresolvedReturnDefault)
   at Prism.DryIoc.Ioc.DryIocContainerExtension.Resolve(Type type)
   at Prism.Ioc.IContainerProviderExtensions.Resolve[T](IContainerProvider provider)
   at Prism.PrismApplicationBase.Initialize()
   at Prism.PrismApplicationBase.InitializeInternal()
   at Dopamine.App.OnStartup(StartupEventArgs e)
   at System.Windows.Application.<.ctor>b__1_0(Object unused)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

Stack trace:
   at System.Windows.Markup.XamlReader.RewrapException(Exception e, IXamlLineInfo lineInfo, Uri baseUri)
   at System.Windows.Markup.WpfXamlLoader.Load(XamlReader xamlReader, IXamlObjectWriterFactory writerFactory, Boolean skipJournaledProperties, Object rootObject, XamlObjectWriterSettings settings, Uri baseUri)
   at System.Windows.Markup.WpfXamlLoader.LoadBaml(XamlReader xamlReader, Boolean skipJournaledProperties, Object rootObject, XamlAccessLevel accessLevel, Uri baseUri)
   at System.Windows.Markup.XamlReader.LoadBaml(Stream stream, ParserContext parserContext, Object parent, Boolean closeStream)
   at Dopamine.Views.MiniPlayer.MiniPlayerPlaylist..ctor(Windows10BorderlessWindow parent, IPlaybackService playbackService, IRegionManager regionManager, IEventAggregator eventAggregator)
   at (Closure\`3 , Windows10BorderlessWindow )
   at Dopamine.Views.Shell.InitializeWindows()
   at Dopamine.Views.Shell..ctor(IContainerProvider container, IWindowsIntegrationService windowsIntegrationService, II18nService i18nService, INotificationService notificationService, IWin32InputService win32InputService, IAppearanceService appearanceService, IPlaybackService playbackService, IMetadataService metadataService, ILifetimeService lifetimeService, IEventAggregator eventAggregator)
   at (Closure\`8 , Object[] , IResolverContext , IScope )
   at DryIoc.Container.ResolveAndCacheDefaultDelegate(Type serviceType, Boolean ifUnresolvedReturnDefault)
   at Prism.DryIoc.Ioc.DryIocContainerExtension.Resolve(Type type)
   at Prism.Ioc.IContainerProviderExtensions.Resolve[T](IContainerProvider provider)
   at Prism.PrismApplicationBase.Initialize()
   at Prism.PrismApplicationBase.InitializeInternal()
   at Dopamine.App.OnStartup(StartupEventArgs e)
   at System.Windows.Application.<.ctor>b__1_0(Object unused)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
Inner Exception 1:
System.ArgumentException: The directory name Dopamine\Playlists is invalid.
   at System.IO.FileSystemWatcher.set_Path(String value)
   at Dopamine.Core.Helpers.GentleFolderWatcher..ctor(String folderPath, Boolean includeSubdirectories, Int32 intervalMilliSeconds)
   at Dopamine.Services.Playlist.PlaylistService..ctor(ITrackRepository trackRepository, IFileService fileService, IContainerProvider container)
   at (Closure\`2 , Object[] , IResolverContext , IScope )
   at DryIoc.SingletonScope.GetOrAddItem(Object[] bucket, Int32 index, CreateScopedValue createValue)
   at DryIoc.Factory.ApplyReuse(Expression serviceExpr, IReuse reuse, Boolean tracksTransientDisposable, Request request)
   at DryIoc.Factory.GetExpressionOrDefault(Request request)
   at DryIoc.Factory.GetDelegateOrDefault(Request request)
   at DryIoc.Container.ResolveAndCacheDefaultDelegate(Type serviceType, Boolean ifUnresolvedReturnDefault)
   at Prism.DryIoc.Ioc.DryIocContainerExtension.Resolve(Type type)
   at Prism.Ioc.IContainerProviderExtensions.Resolve[T](IContainerProvider provider)
   at Dopamine.ViewModels.Common.Base.ContextMenuViewModelBase..ctor(IContainerProvider container)
   at Dopamine.ViewModels.Common.Base.CommonViewModelBase..ctor(IContainerProvider container)
   at Dopamine.ViewModels.Common.Base.TracksViewModelBase..ctor(IContainerProvider container)
   at Dopamine.ViewModels.Common.NowPlayingControlViewModel..ctor(IContainerProvider container)
   at (Object[] , IResolverContext , IScope )
   at DryIoc.Container.ResolveAndCacheDefaultDelegate(Type serviceType, Boolean ifUnresolvedReturnDefault)
   at Prism.DryIoc.Ioc.DryIocContainerExtension.ResolveViewModelForView(Object view, Type viewModelType)
   at Prism.Mvvm.ViewModelLocationProvider.AutoWireViewModelChanged(Object view, Action`2 setDataContextCallback)
   at Prism.Mvvm.ViewModelLocator.AutoWireViewModelChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
   at System.Windows.DependencyObject.OnPropertyChanged(DependencyPropertyChangedEventArgs e)
   at System.Windows.FrameworkElement.OnPropertyChanged(DependencyPropertyChangedEventArgs e)
   at System.Windows.DependencyObject.NotifyPropertyChange(DependencyPropertyChangedEventArgs args)
   at System.Windows.DependencyObject.UpdateEffectiveValue(EntryIndex entryIndex, DependencyProperty dp, PropertyMetadata metadata, EffectiveValueEntry oldEntry, EffectiveValueEntry& newEntry, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType)
   at System.Windows.DependencyObject.SetValueCommon(DependencyProperty dp, Object value, PropertyMetadata metadata, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType, Boolean isInternal)
   at System.Windows.Baml2006.WpfMemberInvoker.SetValue(Object instance, Object value)
   at MS.Internal.Xaml.Runtime.ClrObjectRuntime.SetValue(Object inst, XamlMember property, Object value)

2019-09-10 10:15:34.725|Info|App.xaml.ExecuteEmergencyStop|544|### FORCED STOP of Dopamine, version 2.0.0.1013 ###
2019-09-10 10:15:34.873|Info|Shell.xaml.Window_Closing|124|### STOPPING Dopamine, version 2.0.0.1013 ###
2019-09-10 10:15:34.883|Info|LifetimeService.PerformClosingTasksAsync|25|Performing closing tasks

**Comment by neilunderwood** [comment]
Just want to add this is on my work laptop that uses folder redirection for user profiles.

**Comment by neilunderwood** [comment]
I think I figured this out.  I checked the Program Files(x86)\Dopamine folder for "Playlist" and found it was missing.  I also noticed that the folder was requiring admin credentials to write to.  I changed the permissions on the folder to allow local users, SYSTEM, and Administrators full control of the Program Files(x86)\Dopamine folder and subfolders and everything seems to be working fine now.  Thanks!
